packages/@aws-cdk/aws-codestarnotifications/test/notification-rule.test.ts

```diff
@@ -0,0 +1,196 @@
+      Name: 'MyNotificationRule',
+      DetailType: 'FULL',
+      EventTypeIds: [
+        'codebuild-project-build-state-succeeded',
+        'codebuild-project-build-state-failed',
+      ],
+      Resource: 'arn:aws:codebuild::1234567890:project/MyCodebuildProject',
+      Targets: [
+        {
+          TargetAddress: 'arn:aws:chatbot::1234567890:chat-configuration/slack-channel/MySlackChannel',
+          TargetType: 'AWSChatbotSlack',
+        },
+      ],
+    });
+  });
+
+  test('created new notification rule without name and will generate from the `id`', () => {
+    new notifications.NotificationRule(stack, 'MyNotificationRuleGeneratedFromId', {
+      source: new FakeCodeBuild(),
+    });
+
+    expect(stack).toHaveResourceLike('AWS::CodeStarNotifications::NotificationRule', {
+      Name: 'MyNotificationRuleGeneratedFromId',
+      Resource: 'arn:aws:codebuild::1234567890:project/MyCodebuildProject',
+    });
+  });
+
+  test('generating name will cut if id length is over than 64 charts', () => {
+    const project = new FakeCodeBuild();
+
+    new notifications.NotificationRule(stack, 'MyNotificationRuleGeneratedFromIdIsToooooooooooooooooooooooooooooLong', {
+      source: project,
+    });
+
+    expect(stack).toHaveResourceLike('AWS::CodeStarNotifications::NotificationRule', {
+      Name: 'ificationRuleGeneratedFromIdIsToooooooooooooooooooooooooooooLong',
+      Resource: 'arn:aws:codebuild::1234567890:project/MyCodebuildProject',
+    });
+  });
+
+  test('created new notification rule without detailType', () => {
+    const project = new FakeCodeBuild();
+
+    new notifications.NotificationRule(stack, 'MyNotificationRule', {
+      notificationRuleName: 'MyNotificationRule',
+      source: project,
+    });
+
+    expect(stack).toHaveResourceLike('AWS::CodeStarNotifications::NotificationRule', {
+      DetailType: 'FULL',
+      Name: 'MyNotificationRule',
+      Resource: 'arn:aws:codebuild::1234567890:project/MyCodebuildProject',
+    });
+  });
+
+  test('created new notification rule with status DISABLED', () => {
+    const project = new FakeCodeBuild();
+
+    new notifications.NotificationRule(stack, 'MyNotificationRule', {
+      source: project,
+      enabled: false,
+    });
+
+    expect(stack).toHaveResourceLike('AWS::CodeStarNotifications::NotificationRule', {
+      Name: 'MyNotificationRule',
+      Resource: 'arn:aws:codebuild::1234567890:project/MyCodebuildProject',
+      Status: 'DISABLED',
+    });
+  });
+
+  test('created new notification rule with status ENABLED', () => {
+    const project = new FakeCodeBuild();
+
+    new notifications.NotificationRule(stack, 'MyNotificationRule', {
+      source: project,
+      enabled: true,
+    });
+
+    expect(stack).toHaveResourceLike('AWS::CodeStarNotifications::NotificationRule', {
+      Name: 'MyNotificationRule',
+      Resource: 'arn:aws:codebuild::1234567890:project/MyCodebuildProject',
```

Comment:
Same note about the random hard-coded ARNs being present in test assertions here.

Reply:
Already adjusted.